test/utils/setup.ts

```diff
@@ -132,46 +142,30 @@ export const getSafeTemplate = async (saltNumber: string = getRandomIntAsString(
     return Safe.attach(template) as Safe | SafeL2;
 };
 
-export const getSafeWithOwners = async (
-    owners: string[],
-    threshold: number = owners.length,
-    to: string = AddressZero,
-    data: string = "0x",
-    fallbackHandler: string = AddressZero,
-    logGasUsage: boolean = false,
-    saltNumber: string = getRandomIntAsString(),
-) => {
-    const template = await getSafeTemplate(saltNumber);
+export const getSafeWithOwners = async (safe: safeWithFullConfig) => {
+    if (typeof safe.threshold === "undefined") safe.threshold = safe.owners.length;
+    if (typeof safe.to === "undefined") safe.to = AddressZero;
+    if (typeof safe.data === "undefined") safe.data = "0x";
+    if (typeof safe.fallbackHandler === "undefined") safe.fallbackHandler = AddressZero;
+    if (typeof safe.logGasUsage === "undefined") safe.logGasUsage = false;
+    if (typeof safe.saltNumber === "undefined") safe.saltNumber = getRandomIntAsString();
+
+    const template = await getSafeTemplate(safe.saltNumber);
     await logGas(
-        `Setup Safe with ${owners.length} owner(s)${fallbackHandler && fallbackHandler !== AddressZero ? " and fallback handler" : ""}`,
-        template.setup(owners, threshold, to, data, fallbackHandler, AddressZero, 0, AddressZero),
-        !logGasUsage,
+        `Setup Safe with ${safe.owners.length} owner(s)${safe.fallbackHandler && safe.fallbackHandler !== AddressZero ? " and fallback handler" : ""}`,
+        template.setup(safe.owners, safe.threshold, safe.to, safe.data, safe.fallbackHandler, AddressZero, 0, AddressZero),
+        !safe.logGasUsage,
     );
     return template;
 };
 
-export const getSafeWithSingleton = async (
-    singleton: Safe | SafeL2,
-    owners: string[],
-    threshold?: number,
-    fallbackHandler?: string,
-    saltNumber: string = getRandomIntAsString(),
-) => {
+export const getSafeWithSingleton = async (singleton: Safe | SafeL2, owners: string[], saltNumber: string = getRandomIntAsString()) => {
```

Comment:
To me, one of the main points of this refactor is to have a single function instead of multiple:

```ts
export const getSafe(options: ...)
```

This way, `getSafeWithOwners(owners)` => `getSafe({ owners })`, and `getSafeWithSigleton(singleton, owners)` => `getSafe({ singleton, owners })`.

Reply:
@remedcu - if you want to implement this in a follow-up PR, feel free to do so. It would "unify Safe creation methods", and can arguably be split out.